core/src/test/java/com/linecorp/armeria/common/metric/NettyMetricsTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright 2023 LINE Corporation
+ *
+ *  LINE Corporation licenses this file to you under the Apache License,
+ *  version 2.0 (the "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at:
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package com.linecorp.armeria.common.metric;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.util.concurrent.CountDownLatch;
+
+import org.junit.Test;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import io.netty.channel.DefaultEventLoopGroup;
+import io.netty.channel.EventLoopGroup;
+
+public class NettyMetricsTest {
+
+    private class BlockMe extends CountDownLatch implements Runnable {
+
+        BlockMe() {
+            super(1);
+        }
+
+        @Override
+        public void run() {
+            try {
+                await();
+            } catch (Throwable ignored) {
+            }
+        }
+    }
+
+    @Test
+    public void test() {
+        final MeterRegistry registry = new SimpleMeterRegistry();
+        final NettyMetrics.Self metrics =
+                new NettyMetrics.Self(
+                        registry,
+                        new MeterIdPrefix("foo")
+                );
+
+        final BlockMe task = new BlockMe();
+
+        final EventLoopGroup workers = new DefaultEventLoopGroup(2);
+        // Block both executors
+        workers.submit(task);
+        workers.submit(task);
+
+        workers.submit(() -> {});
+
+        metrics.add(workers);
+
+        // Check that API works as expected
+        assertThat(metrics.pendingTasks()).isEqualTo(1.0);
```

Comment:
This condition looks flaky. The `task`s submitted may not be scheduled here.
```
 NettyMetricsTest > test FAILED
    org.opentest4j.AssertionFailedError: 
    expected: 1.0
     but was: 2.0
        at java.base@19.0.2/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:67)
        at java.base@19.0.2/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
        at app//com.linecorp.armeria.common.metric.NettyMetricsTest.test(NettyMetricsTest.java:69)
```
How about adding `AtomicInteger counter` to `BlockMe`?
```java
// Block both executors
workers.submit(task);
workers.submit(task);
// Make sure that all tasks are scheduled.
await().untilAtomic(task.counter, Matchers.equalTo(2));

workers.submit(() -> {});
assertThat(metrics.pendingTasks()).isEqualTo(1.0);
```